### ISSUE TITLE
Exception handling in DataCollatorForCompletionOnlyLM

### ISSUE THREAD

**Issue by tannonk** [huggingface/trl#643]
Hey,

While testing out the new `DataCollatorForCompletionOnlyLM` on custom data, I had some issues relating to the exception handling of batch instances in which we fail to find the `response_template` or `instruction_template` (https://github.com/huggingface/trl/blob/3b2c820db68aecca36c2ff5d152de59deec8fe5c/trl/trainer/utils.py#L142C3-L142C3). 

It turned out that I had some exceptionally long responses in my training data which didn't actually contain both a valid `response_template` and `instruction_template`.

**Possible fix:**

My current work around was to replace the `RuntimeError` with a more informative `warning` and simply set the labels for a problematic instance to `self.ignore_index`, so that it will be ignored for the loss calculation but training will still run successfully (provided this doesn't happen for every instance 😉 ), e.g.:

```python
if len(human_token_ids_idxs) == 0:
    warnings.warn(
        f'Could not find instruction key `{self.instruction_template}` in the following instance {self.tokenizer.decode(batch["input_ids"][i])} ' \
        f'This instance will be ignored in loss calculation.'
    )
    batch["labels"][i, :] = self.ignore_index
```

Would something like this be more suitable than the current exception handling?

**Comment by lvwerra** [comment]
Thanks for the investigation - that looks like a good fix to me! Would you like to open a PR?
